crates/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -614,7 +627,7 @@ impl<'a> FunctionContext<'a> {
     /// `extract_current_value` for more details.
     pub(super) fn assign_new_value(&mut self, lvalue: LValue, new_value: Values) {
         match lvalue {
-            LValue::Ident(references) => self.assign(references, new_value),
+            LValue::Ident => unreachable!("Cannot assign to a variable without a reference"),
```

Comment:
This is possible now because mutable variables return a `LValue::Dereference` instead